################################################################################
SHORT DESCRIPTION: 
################################################################################
Use large dimensions for tablet absolute coordinates.

################################################################################
LONG DESCRIPTION:
################################################################################

################################################################################
CHANGELOG 
################################################################################
Original author: unknown
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>
Port to 4.14: Richard Turner <turnerr@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
?

################################################################################
UPSTREAM PLAN
################################################################################
None.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES 
################################################################################
--- a/drivers/input/misc/xen-kbdfront.c
+++ b/drivers/input/misc/xen-kbdfront.c
@@ -303,9 +303,9 @@ static int xenkbd_probe(struct xenbus_de
 		if (abs) {
 			__set_bit(EV_ABS, ptr->evbit);
 			input_set_abs_params(ptr, ABS_X, 0,
-					     ptr_size[KPARAM_X], 0, 0);
+					     XENFB_TABLET_WIDTH, 0, 0);
 			input_set_abs_params(ptr, ABS_Y, 0,
-					     ptr_size[KPARAM_Y], 0, 0);
+					     XENFB_TABLET_HEIGHT, 0, 0);
 		} else {
 			input_set_capability(ptr, EV_REL, REL_X);
 			input_set_capability(ptr, EV_REL, REL_Y);
--- a/include/xen/interface/io/fbif.h
+++ b/include/xen/interface/io/fbif.h
@@ -137,6 +137,8 @@ struct xenfb_page {
 #ifdef __KERNEL__
 #define XENFB_WIDTH 800
 #define XENFB_HEIGHT 600
+#define XENFB_TABLET_WIDTH 32767
+#define XENFB_TABLET_HEIGHT 32767
 #define XENFB_DEPTH 32
 #endif
 
